includes/import/class-wc-product-csv-importer.php

```diff
@@ -976,7 +976,7 @@ public function import() {
 				continue;
 			}
 
-			if ( $update_existing && ( $id || $sku ) && ! $id_exists && ! $sku_exists ) {
+			if ( $update_existing && ( isset( $parsed_data['id'] ) || isset( $parsed_data['sku'] ) ) && ! $id_exists && ! $sku_exists ) {
```

Comment:
Is the test `( isset( $parsed_data['id'] ) || isset( $parsed_data['sku'] ) )` needed? It seems to me that only checking if `! $id_exists && ! $sku_exists` is enough or am I missing something?

Reply:
We only want to skip based on these values if they actually provided an ID or SKU. if they did not, we should import the product as new.

Reply:
Maybe I'm missing something, but it seems to me that when `$update_existing` is true we ignore products that don't exist already. In other words, if ID or SKU are not provided and `$update_existing` is true, we want to skip the product instead of creating a new one. At least that is my understanding from the description of the "Update existing products" checkbox: "Existing products that match by ID or SKU will be updated. Products that do not exist will be skipped."

Reply:
Thats correct @rodrigoprimo, but `! $id_exists && ! $sku_exists` would also be true if you left ID and SKU empty. It doesn't match an existing product and should import in that case.